<h1>Table of Contents<span class="tocSkip"></span></h1>
<div class="toc"><ul class="toc-item"></ul></div>

In [1]:
import numpy as np
import pandas as pd
from statsmodels.stats.weightstats import _tconfint_generic as t_stat
from statsmodels.stats.weightstats import _zconfint_generic as z_stat

**1 Задание**

**Даны значения величины заработной платы заемщиков банка (zp) и значения их поведенческого кредитного скоринга (ks):**

**zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110]**

**ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832].**

**а) Найдите ковариацию этих двух величин с помощью элементарных действий, а затем с помощью функции cov из numpy. Полученные значения должны быть равны.**

$M(zp)=\frac{35+45+190+200+40+70+54+150+120+110}{10}=101.4$

$M(ks)=\frac{401+574+874+919+459+739+653+902+746+832}{10}=709.9$

$cov(zp, kz)=((35-101.4)(401-709.9)+(45-101.4)(574-709.9)+(190-101.4)(874-709.9)+$

$+(200-101.4)(919-709.9)+(40-101.4)(459-709.9)+(70-101.4)(739-709.9)+$

$+(54-101.4)(653-709.9)+(150-101.4)(902-709.9)+(120-101.4)(746-709.9)+$

$+(110-101.4)(832-709.9))/(10-1)=$

$(20510.96+7664.76+14539.26+20617.26+15405.26+913.74+2697.06+9336.06+671.46+1050.06)/9=10175.37(7)$

решение на Python

In [2]:
zp = np.array([35, 45, 190, 200, 40, 70, 54, 150, 120, 110])
ks = np.array([401, 574, 874, 919, 459, 739, 653, 902, 746, 832])
cov = np.cov(zp, ks)
np.round(cov[1, 0], 4)

10175.3778

**б) Найдите коэффициент корреляции Пирсона с помощью ковариации и среднеквадратичных отклонений двух признаков, а затем с использованием функций из библиотек numpy и pandas.**

$D(zp)=\frac{1}{n-1}\sum\limits_{i=1}^{n}(zp_i-M(zp))^2=((32-101.4)^2+(32-101.4)^2+(32-101.4)^2+(32-101.4)^2+$

$+(32-101.4)^2+(32-101.4)^2+(32-101.4)^2+(32-101.4)^2+(32-101.4)^2+(32-101.4)^2)/(10-1)=$

$=(4408.96+3180.96+7849.96+9721.96+3769.96+985.96+2246.76+2361.96+345.96)/9=34872.44/9=3847.71$

$\sigma_{zp}=\sqrt{D(zp)}=\sqrt{3847.71}=62.2472$

$D(ks)=\frac{1}{n-1}\sum\limits_{i=1}^{n} (ks_i-M(ks))^2=((401-709.9)^2+(574-709.9)^2+(874-709.9)^2+(919-709.9)^2+$

$+(459-709.9)^2+(739-709.9)^2+(653-709.9)^2+(902-709.9)^2+(746-709.9)^2+(832-709.9)^2)/(10-1)=$

$=(95419.21+18468.81+26928.81+43722.81+62950.81+846.81+3237.61+36902.41+1303.21+14908.41)/9=$

$=304688.90/9=33854.32(2)$

$\sigma_{ks}=\sqrt{D(ks)}=\sqrt{33854.32}=183.9955$

$r_{zp, ks}=\frac{cov(zp, ks)}{\sigma_{zp}\cdot \sigma_{ks}}=\frac{10175.38}{62.2472 \cdot 183.9955}=\frac{10175.38}{11453.2047}=0.8884$

решение на Python

In [3]:
np.round(cov[1, 0]/(zp.std(ddof=1)*ks.std(ddof=1)), 4)

0.8875

**2 Задание**

**Измерены значения IQ выборки студентов, обучающихся в местных технических вузах: 131, 125, 115, 122, 131, 115, 107, 99, 125, 111. Известно, что в генеральной совокупности IQ распределен нормально. Найдите доверительный интервал для математического ожидания с надежностью 0.95.**

неизвестна $\sigma \Rightarrow$ не было анализа генеральной совокупности $\Rightarrow$ критерием оценки будет критерий Стьюдента $t$ из таблицы Стьюдента

тогда ошибка среднего - это $\frac{s}{\sqrt{n}}$, где $s$ - корень дисперсии по выборке $s=\sqrt{D_в(X)}$, а $\sqrt{D_в(X)}$ - несмещённая оценка дисперсии

генеральная совокупность распределена нормально $\Rightarrow$ можем перейти от $\alpha$ к $\frac{\alpha}{2}$

$p=0.95 \Rightarrow \alpha=1-p=1-0.95=0.05$

$\frac{\alpha}{2} = \frac{0.05}{2}=0.025$

считаем среднюю величину по выборке

$\overline{X_в}=\frac{131+125+115+122+131+115+107+99+125+111}{10}=\frac{1181}{10}=181.1$

Находим дисперсию $D_в(X)$, а из неё среднеквадратичное отклонение $\sigma$

$D_в(X)=\frac{1}{n-1}\sum\limits_{i=1}^{n}(x_i-\overline{x_в})^2 = ((131-118.1)^2+(125-118.1)^2+(115-118.1)^2+(122-118.1)^2+$

$+(131-118.1)^2+(115-118.1)^2+(107-118.1)^2+(99-118.1)^2+(125-118.1)^2+(111-118.1)^2)/(10-1)=$

$(166.41+47.61+9.61+15.21+166.41+9.61+123.21+364.81+47.61+50.41)/9=1000.9/9=111.21(1)$

$\sigma=\sqrt{D_в(X)}=\sqrt{111.21}=10.5457$

находим табличное значение $t$ Стьюдента, а для него:

$k=n-1=10-1=9$

$p=1-\frac{\alpha}{2}=1-0.025=0.975$

$\Rightarrow t_{\frac{\alpha}{2}}=2.262$

находим доверительный интервал по формуле:

$\left[
  \begin{array}{ccc}
     \overline{X_в} \mp t_{\frac{\alpha}{2}}\cdot \frac{s}{\sqrt{n}}\\
  \end{array}
\right]
\Rightarrow
\left[
  \begin{array}{ccc}
     118.1 \mp 2.262\cdot \frac{10.5457}{\sqrt{10}}\\
  \end{array}
\right]
\Rightarrow
\left[
  \begin{array}{ccc}
     118.1-2.262\cdot3.3348; 118.1+2.262\cdot3.3348\\
  \end{array}
\right]
$

$\left[
  \begin{array}{ccc}
     110.5567; 125.6433\\
  \end{array}
\right]
-$ этот интервал покрывает истинное значение $X$ с доверительной вероятностью 95%

решение на Python

In [4]:
x = np.array([131, 125, 115, 122, 131, 115, 107, 99, 125, 111])
result = t_stat(x.mean(), x.std(ddof=1)/(np.sqrt(len(x))), len(x) - 1, 0.05, 'two-sided')
print(f'[{np.round(result[0], 4)} : {np.round(result[1], 4)}]')

[110.5561 : 125.6439]


**3 Задание**

**Известно, что рост футболистов в сборной распределен нормально
с дисперсией генеральной совокупности, равной 25 кв.см. Объем выборки равен 27, среднее выборочное составляет 174.2. Найдите доверительный интервал для математического ожидания с надежностью 0.95.**

до нас был выполнен анализ генеральной совокупности $\Rightarrow$ критерием оценки будет критерий Фишера $Z$ из таблицы распределения Лапласа для $Z<0$ (так как мы работаем с доверительным интервалом)

генеральная совокупность распределена нормально $\Rightarrow$ можем перейти от $\alpha$ к $\frac{\alpha}{2}$

$p=0.95 \Rightarrow \alpha=1-p=1-0.95=0.05$

$\frac{\alpha}{2} = \frac{0.05}{2}=0.025$

так как был анализ генеральной совокупности, то стандартной ошибкой среднего будет $\frac{\sigma}{\sqrt{n}}$, где $\sigma$ взята из генеральной совокупности

Находим табличное значение $Z$

$Z(P_{\forall Z}=0.025)=-1.96$

находим доверительный интервал по формуле:

$\left[
  \begin{array}{ccc}
     \overline{X_в} \mp Z_{\frac{\alpha}{2}}\cdot \frac{\sigma}{\sqrt{n}}\\
  \end{array}
\right]
\Rightarrow
\left[
  \begin{array}{ccc}
     174.2 \mp 1.96\cdot \frac{5}{\sqrt{27}}\\
  \end{array}
\right]
$

$\left[
  \begin{array}{ccc}
     172.3140; 176.0860\\
  \end{array}
\right]
-$ этот интервал покрывает истинное значение $M(X)$ генеральной совокупности с вероятностью 95%

решение на Python

In [5]:
result = z_stat(174.2, np.sqrt(25)/np.sqrt(27), 0.05, 'two-sided')
print(f'[{np.round(result[0], 4)} : {np.round(result[1], 4)}]')

[172.314 : 176.086]
